src/Razor/src/rzls/rzls.csproj

```diff
@@ -1,36 +1,90 @@
+    this and extract the
+        executables when it builds.  NuGet just happens to be a convenient way to share build artifacts.
+    -->
+    <IsPackable>true</IsPackable>
+    <!-- Our outer Pack task (defined in PackAllRids.targets) invokes Pack passing in a
+    PackRuntimeIdentifier to produce one package per RID; from that we can set everything else. -->
+    <RuntimeIdentifier Condition="'$(PackRuntimeIdentifier)' != '' and '$(PackRuntimeIdentifier)' != 'neutral'">$(PackRuntimeIdentifier)</RuntimeIdentifier>
+
+    <!--
+        By default since this is an Exe project and we build on windows, we'll get a .exe as output from a
+    platform neutral build.
+        However, we really only want an executable if we're building for a specific platform (aka have a
+    runtime identifier).
+
+        So if we don't have a platform, tell the build not to output a .exe file because we're building
+    platform neutral bits.
+    -->
+    <UseAppHost Condition="'$(RuntimeIdentifier)' == ''">false</UseAppHost>
+
+    <PackageId>$(AssemblyName).$(PackRuntimeIdentifier)</PackageId>
 
-  <PropertyGroup Condition="'$(DotNetBuild)' == 'true'">
-    <RuntimeIdentifiers>$(TargetRid)</RuntimeIdentifiers>
-    <RuntimeIdentifier>$(TargetRid)</RuntimeIdentifier>
+    <!--
+        Publish the platform specific executables before any of the pack related targets run.
+        This is important to ensure that the content files exist on the project before nuget looks for
+    them.
+
+        Note that BeforeTargets="Pack" is not enough, as the determination of what to include in the
+    package runs
+        before the actual "Pack" target runs.
+    -->
+    <BeforePack>PackPublishContent;$(BeforePack)</BeforePack>
+
+    <!--
+      NU5100: Since this is not a standard nuget package and is only consumed by the extension build, we
+    don't care if the folder structure doesn't match what nuget expects.
+      NETSDK1206: Until we move to a newer release of the SQLitePCLRaw packages (2.1.6+) that removes the
+    libs for alpine RIDs, we have to ignore the SDK warning about distribution specific RIDs.
+    -->
+    <NoWarn>$(NoWarn);NU5100;NETSDK1206</NoWarn>
+
+    <!-- Publishing settings -->
     <SelfContained>false</SelfContained>
-    <AppendRuntimeIdentifierToOutputPath>false</AppendRuntimeIdentifierToOutputPath>
-    <RidSpecificPublishNoBuildProperty>NoBuild=true</RidSpecificPublishNoBuildProperty>
+    <PublishDir Condition="'$(RuntimeIdentifier)' != ''">$(ArtifactsDir)LanguageServer\$(Configuration)\$(TargetFramework)\$(RuntimeIdentifier)</PublishDir>
+    <PublishDir Condition="'$(RuntimeIdentifier)' == ''">$(ArtifactsDir)LanguageServer\$(Configuration)\$(TargetFramework)\neutral</PublishDir>
+
+    <!-- List of runtime identifiers that we want to publish an executable for. -->
+    <!-- When building a VMR vertical, we don't need to pack everything. Just pack the passed in
```

Comment:
I copied this, but I don't think razor is part of VMR? We might be able to remove this

Reply:
We are: https://github.com/dotnet/dotnet/tree/main/src/razor

Though I admit I'm not sure if rzls specifically needs to be, but apparently it is.